File: src/Factory.php
<?php

namespace Lab_Web;

use Lab_Web\Model\CompModelImpl;
use Lab_Web\Model\MainModelImpl;

final class Factory {

    /**
     * @throws \Exception always
     */
    public function __construct() {
        throw new \Exception("Operation is not implemented");
    }

    public static function makeCompModel($xes = null, $y = null, $r = null) {
        if (!is_array($xes)) {
            $newXes = null;

            if (key_exists('x', $_GET)) {
                $newXes = $_GET['x'];

                if (!is_array($newXes)) {
                    $newXes = [$newXes];
                }

                foreach ($newXes as &$x) {
                    try {
                        $x = Utility::toFloat($x);
                    } catch (\InvalidArgumentException $e) {}
                }
            }

            if (is_array($newXes)) {
                $xes = $newXes;
            }
        }

        if (!is_numeric($y)) {
            try {
                $newY = null;

                if (key_exists('y', $_GET)) {
                    $newY = Utility::toFloat($_GET['y']);
                }

                if (is_numeric($newY)) {
                    $y = $newY;
                }
            } catch (\InvalidArgumentException $e) {
                $y = null;
            }
        }

        if (!is_numeric($r)) {
            try {
                $newR = null;

                if (key_exists('r', $_GET)) {
                    $newR = Utility::toFloat($_GET['r']);
                }

                if (is_numeric($newR)) {
                    $r = $newR;
                }
            } catch (\InvalidArgumentException $e) {
                $r = null;
            }
        }

        return new CompModelImpl($xes, $y, $r);
    }

    public static function makeMainModel($doFrontendTimeUpdate = null, $startTime = null) {
        $model = new MainModelImpl($startTime);

        if (!is_bool($doFrontendTimeUpdate)) {
            $newDoFrontendTimeUpdate = null;

            if (key_exists('doFrontendTimeUpdate', $_COOKIE)) {
                $newDoFrontendTimeUpdate = $_COOKIE['doFrontendTimeUpdate'];
            }

            if (key_exists('doFrontendTimeUpdate', $_GET)) {
                $newDoFrontendTimeUpdate = $_GET['doFrontendTimeUpdate'];
            }

            $doFrontendTimeUpdate = Utility::toBoolean($newDoFrontendTimeUpdate, $doFrontendTimeUpdate);
        }

        if (is_bool($doFrontendTimeUpdate)) {
            $model->setDoFrontendTimeUpdate($doFrontendTimeUpdate);
        }

        return $model;
    }
}

File: src/ru/byprogminer/Lab2_Web/utility/Factory.java
package ru.byprogminer.Lab2_Web.utility;

import ru.byprogminer.Lab2_Web.model.CompModel;
import ru.byprogminer.Lab2_Web.model.CompModelImpl;
import ru.byprogminer.Lab2_Web.model.MainModel;
import ru.byprogminer.Lab2_Web.model.MainModelImpl;

import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletRequest;
import java.math.BigDecimal;
import java.util.Arrays;
import java.util.Optional;

public class Factory {

    private final HttpServletRequest request;

    public Factory(HttpServletRequest request) {
        this.request = request;
    }

    public MainModel makeMainModel() {
        return makeMainModel(null);
    }

    public MainModel makeMainModel(Boolean doFrontendTimeUpdate) {
        return makeMainModel(doFrontendTimeUpdate, null);
    }

    public MainModel makeMainModel(Boolean doFrontendTimeUpdate, Long startTime) {
        final MainModelImpl model = new MainModelImpl(startTime);

        if (doFrontendTimeUpdate == null) {
            String newDoFrontendTimeUpdate = null;

            final Cookie cookie = Arrays.stream(Optional.ofNullable(request.getCookies()).orElse(new Cookie[] {}))
                    .filter(c -> c.getName().equals("doFrontendTimeUpdate"))
                    .findFirst().orElse(null);
            if (cookie != null) {
                newDoFrontendTimeUpdate = cookie.getValue();
            }

            final String param = request.getParameter("doFrontendTimeUpdate");
            if (param != null) {
                newDoFrontendTimeUpdate = param;
            }

            doFrontendTimeUpdate = Utility.parseBoolean(newDoFrontendTimeUpdate, doFrontendTimeUpdate);
        }

        if (doFrontendTimeUpdate != null) {
            model.setDoFrontendTimeUpdate(doFrontendTimeUpdate);
        }

        return model;
    }

    public CompModel makeCompModel() {
        return makeCompModel(null, null, null);
    }

    public CompModel makeCompModel(BigDecimal x, BigDecimal y, BigDecimal r) {
        if (x == null) {
            try {
                BigDecimal newX = null;

                final String param = request.getParameter("x");
                if (param != null) {
                    newX = new BigDecimal(param);
                }

                x = newX;
            } catch (Throwable e) {
                e.printStackTrace();
            }
        }

        if (y == null) {
            try {
                BigDecimal newY = null;

                final String param = request.getParameter("y");
                if (param != null) {
                    newY = new BigDecimal(param);
                }

                y = newY;
            } catch (Throwable e) {
                e.printStackTrace();
            }
        }

        if (r == null) {
            try {
                BigDecimal newR = null;

                final String param = request.getParameter("r");
                if (param != null) {
                    newR = new BigDecimal(param);
                }

                r = newR;
            } catch (Throwable e) {
                e.printStackTrace();
            }
        }

        return new CompModelImpl(x, y, r);
    }
}
